### DISCUSSION TITLE
Is it compatible with Citus?

### DISCUSSION THREAD

**Discussion by suvishodcitus** [osm2pgsql-dev/osm2pgsql#1926] [Q&A]
We are trying to use it with Citus DB (distributed PG), but its dropping already distributed tables and recreating them.
Is there a way to run distribution commands in a post script?

**Comment by joto** [comment]
> We are trying to use it with Citus DB (distributed PG), but its dropping already distributed tables and recreating them.

I don't now Citus, but from your description it could have something to do with the (pseudo-)clustering feature of osm2pgsql. By default osm2pgsql first creates UNLOGGED tables, then later sorts the data and copies it into a new table. The temporary table is removed and the new table renamed. It might be that Citus doesn't work well with that UNLOGGED table. You can disable this feature by setting `cluster = 'no'` in your table definitions in the flex output. See https://osm2pgsql.org/doc/manual.html#clustering-by-geometry .

> Is there a way to run distribution commands in a post script?

I don't know that "distribution commands" are. If you simply want to run a command after osm2pgsql has run, wrap it in a shell script and run the command from there. If you need something else, describe in more detail what you need.
